### PR TITLE
[Core] enable put small object in memory store

### PR DESCRIPTION
**Pull request by zhuohan123**
## Why are these changes needed?

Previously we disabled putting small objects into the memory store by default. We turn this on by default in this PR.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by rkooo567** [comment]
cc @raulchen

**Review by suquark — APPROVED** [review]
LGTM. We should merge after all related tests pass.

**Comment by suquark** [comment]
@zhuohan123 could you have a review that whether there are related CI failures?

**Comment by suquark** [comment]
@zhuohan123 could you update this PR?

File: src/ray/common/ray_config_def.h [hunk 1]
```diff
@@ -322,10 +322,10 @@ RAY_CONFIG(int64_t, metrics_report_batch_size, 100)
...
-/// Whether start the Plasma Store as a Raylet thread.
-RAY_CONFIG(bool, put_small_object_in_memory_store, false)
+/// Whether to put small objects directly into the memory store.
+RAY_CONFIG(bool, put_small_object_in_memory_store, true)
 
 /// Maximum number of tasks that can be in flight between an owner and a worker for which
 /// the owner has been granted a lease. A value >1 is used when we want to enable
 /// pipelining task submission.
-RAY_CONFIG(uint32_t, max_tasks_in_flight_per_worker, 1)
+RAY_CONFIG(uint32_t, max_tasks_in_flight_per_worker, 1)
```
Comment (suquark) [inline src/ray/common/ray_config_def.h:331]: don't remove the last empty line

**Review by suquark — CHANGES_REQUESTED** [review]

**Comment by raulchen** [comment]
Hi @zhuohan123 @suquark , do you know what are missing in Java to turn this flag on?

**Comment by zhuohan123** [comment]
> Hi @zhuohan123 @suquark , do you know what are missing in Java to turn this flag on?

Some Java tests are failing when we turn this flag on, so we turn this off temporarily to pass the tests. I will take a look later to fix the Java tests.

In general, turning this flag on will cause some memory limits tests to fail. For example if we set the memory limit to 10M and put a 10M object. If now we put a very small object like a string "HI", previously there will be an OOM error, but now since the string will be put into the memory store, the OOM error will not exist anymore. I'm not sure whether there are these kinds of tests in Java for now but will check later.

**Pull request closed without merging**